kernel/src/net/socket/ip/raw/mod.rs

```diff
@@ -0,0 +1,427 @@
+// SPDX-License-Identifier: MPL-2.0
+
+use core::sync::atomic::{AtomicBool, Ordering};
+
+use aster_bigtcp::{
+    socket::{SocketEventObserver, SocketEvents},
+    wire::{IpAddress, IpEndpoint},
+};
+use ostd::sync::LocalIrqDisabled;
+use takeable::Takeable;
+
+use self::{bound::BoundRaw, unbound::UnBoundRaw};
+use super::{common::get_ephemeral_endpoint, UNSPECIFIED_LOCAL_ENDPOINT};
+use crate::{
+    events::IoEvents,
+    fs::{
+        file_handle::FileLike,
+        utils::{InodeMode, Metadata, StatusFlags},
+    },
+    match_sock_option_mut,
+    net::{
+        iface::IfaceEx,
+        socket::{
+            options::{Error as SocketError, IpHdrIncl, SocketOption},
+            util::{
+                options::{IpSocketOptionSet, SocketOptionSet},
+                send_recv_flags::SendRecvFlags,
+                socket_addr::SocketAddr,
+                MessageHeader,
+            },
+            Socket,
+        },
+    },
+    prelude::*,
+    process::signal::{PollHandle, Pollable, Pollee},
+    util::{net::Protocol, MultiRead, MultiWrite},
+};
+
+mod bound;
+mod unbound;
+
+#[derive(Debug, Clone)]
+struct OptionSet {
+    socket: SocketOptionSet,
+    ip: IpSocketOptionSet,
+}
+
+impl OptionSet {
+    fn new() -> Self {
+        let socket = SocketOptionSet::new_raw();
+        let ip = IpSocketOptionSet::new_raw();
+        OptionSet { socket, ip }
+    }
+}
+
+pub struct RawDGramSocket {
+    options: RwLock<OptionSet>,
+    inner: RwLock<Takeable<Inner>, LocalIrqDisabled>,
+    nonblocking: AtomicBool,
```

Comment:
I know this code is copied from UDP sockets and they name the variable as `nonblocking`. But please name this variable as `is_nonblocking` is new code, which is a better name since `is_xxx` makes it clear that it's a boolean variable.